In [13]:
import pandas as pd
import matplotlib.pyplot as plt
pd.set_option('display.max_rows', None)  # or set a large number instead of None
pd.set_option('display.max_columns', None)  # to show all columns

In [14]:
df_weather = pd.read_csv('./data/combine_final_weather.csv')
df_weather.info()


/var/folders/sg/hlv7r8p13tj6dbrhj1phzxlm0000gn/T/ipykernel_70228/3640038700.py:1: DtypeWarning: Columns (14,20,21,24,25,26,27,28,32,34) have mixed types. Specify dtype option on import or set low_memory=False.
  df_weather = pd.read_csv('./data/combine_final_weather.csv')


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6361912 entries, 0 to 6361911
Data columns (total 39 columns):
 #   Column                  Dtype  
---  ------                  -----  
 0   id                      object 
 1   type                    object 
 2   coordinates             object 
 3   type.1                  object 
 4   STATION_PRESSURE        float64
 5   WINDCHILL_FLAG          float64
 6   UTC_DAY                 int64  
 7   STATION_NAME            object 
 8   HUMIDEX_FLAG            float64
 9   WIND_SPEED              float64
 10  VISIBILITY              float64
 11  DEW_POINT_TEMP          float64
 12  UTC_DATE                object 
 13  HUMIDEX                 float64
 14  WIND_SPEED_FLAG         object 
 15  UTC_YEAR                int64  
 16  LOCAL_DAY               int64  
 17  RELATIVE_HUMIDITY       float64
 18  PROVINCE_CODE           object 
 19  LOCAL_HOUR              int64  
 20  CLIMATE_IDENTIFIER      object 
 21  VISIBILITY_FLAG         object 

In [ ]:
"""
severe_wind_days = combined_df[combined_df['WIND_SPEED']
                               > WIND_SPEED_THRESHOLD]

winter_storm_days = combined_df[(combined_df['type'] == 'snow') & (
    combined_df['PRECIP_AMOUNT'] >= SNOW_PRECIP_THRESHOLD)]

rainny_storm = combined_df[(combined_df['type'] == 'rain') & (combined_df['PRECIP_AMOUNT'] >= RAIN_PRECIP_THRESHOLD)]

snow_severe = df_weather[(df_weather['WEATHER_ENG_DESC'].str.contains('snow')) & (
    df_weather['PRECIP_AMOUNT'] >= SNOW_PRECIP_THRESHOLD)]
rain_severe = df_weather[(df_weather['WEATHER_ENG_DESC'].str.contains(
    'rain')) & (df_weather['PRECIP_AMOUNT'] >= RAIN_PRECIP_THRESHOLD)]
"""

In [15]:

df_weather['UTC_DATE'] = pd.to_datetime(df_weather['UTC_DATE'], utc=True)
df_weather['LOCAL_DATE'] = df_weather['UTC_DATE'].dt.tz_convert('America/New_York')
df_weather['LOCAL_DATE'] = df_weather['LOCAL_DATE'].dt.strftime('%Y-%m-%d')
df_weather.head()


id     type                  coordinates type.1  \
0  6106398.2018.2.12.17  Feature  [-77.31666666666666, 45.95]  Point   
1   6106398.2018.1.19.9  Feature  [-77.31666666666666, 45.95]  Point   
2  6106398.2018.1.19.10  Feature  [-77.31666666666666, 45.95]  Point   
3  6106398.2018.1.19.11  Feature  [-77.31666666666666, 45.95]  Point   
4  6106398.2018.1.30.15  Feature  [-77.31666666666666, 45.95]  Point   

   STATION_PRESSURE  WINDCHILL_FLAG  UTC_DAY STATION_NAME  HUMIDEX_FLAG  \
0            101.84             NaN       12   PETAWAWA A           NaN   
1             99.44             NaN       19   PETAWAWA A           NaN   
2             99.44             NaN       19   PETAWAWA A           NaN   
3             99.42             NaN       19   PETAWAWA A           NaN   
4            100.76             NaN       30   PETAWAWA A           NaN   

   WIND_SPEED  VISIBILITY  DEW_POINT_TEMP                  UTC_DATE  HUMIDEX  \
0        11.0        24.1           -16.5 2018-02-12 22:00:00+00:00      NaN   
1         9.0         4.8            -6.7 2018-01-19 14:00:00+00:00      NaN   
2        11.0         4.8            -6.8 2018-01-19 15:00:00+00:00      NaN   
3        11.0         4.8            -6.4 2018-01-19 16:00:00+00:00      NaN   
4        15.0        24.1           -18.5 2018-01-30 20:00:00+00:00      NaN   

  WIND_SPEED_FLAG  UTC_YEAR  LOCAL_DAY  RELATIVE_HUMIDITY PROVINCE_CODE  \
0             NaN      2018         12               47.0            ON   
1             NaN      2018         19               85.0            ON   
2             NaN      2018         19               81.0            ON   
3             NaN      2018         19               81.0            ON   
4             NaN      2018         30               57.0            ON   

   LOCAL_HOUR CLIMATE_IDENTIFIER VISIBILITY_FLAG  TEMP  UTC_MONTH  \
0          17            6106398             NaN  -7.0          2   
1           9            6106398             NaN  -4.5          1   
2          10            6106398             NaN  -4.1          1   
3          11            6106398             NaN  -3.6          1   
4          15            6106398             NaN -11.6          1   

  STATION_PRESSURE_FLAG WIND_DIRECTION_FLAG PRECIP_AMOUNT_FLAG  \
0                   NaN                 NaN                NaN   
1                   NaN                 NaN                NaN   
2                   NaN                 NaN                NaN   
3                   NaN                 NaN                NaN   
4                   NaN                 NaN                NaN   

       WEATHER_ENG_DESC DEW_POINT_TEMP_FLAG  LOCAL_DATE  PRECIP_AMOUNT  \
0          Mainly Clear                 NaN  2018-02-12            NaN   
1  Freezing Drizzle,Fog                 NaN  2018-01-19            NaN   
2  Freezing Drizzle,Fog                 NaN  2018-01-19            NaN   
3              Snow,Fog                 NaN  2018-01-19            NaN   
4                 Clear                 NaN  2018-01-30            NaN   

                WEATHER_FRE_DESC TEMP_FLAG  LOCAL_YEAR RELATIVE_HUMIDITY_FLAG  \
0            Généralement dégagé       NaN        2018                    NaN   
1  Bruine verglaçante,Brouillard       NaN        2018                    NaN   
2  Bruine verglaçante,Brouillard       NaN        2018                    NaN   
3               Neige,Brouillard       NaN        2018                    NaN   
4                         Dégagé       NaN        2018                    NaN   

                     ID  WINDCHILL  LOCAL_MONTH  WIND_DIRECTION  
0  6106398.2018.2.12.17      -12.0            2            31.0  
1   6106398.2018.1.19.9       -8.0            1            13.0  
2  6106398.2018.1.19.10       -9.0            1            13.0  
3  6106398.2018.1.19.11       -8.0            1            12.0  
4  6106398.2018.1.30.15      -19.0            1            33.0

In [ ]:
# Extract the hour component
df_weather['LOCAL_HOUR'] = df_weather['UTC_DATE'].dt.hour


# Convert UTC_DATE to date-only format
df_weather['UTC_DATE'] = df_weather['UTC_DATE'].dt.date



In [19]:
df_weather_hour = df_weather.groupby(['LOCAL_DATE', 'LOCAL_HOUR'])
df_weather_hour['LOCAL_HOUR'].count().head(10)

LOCAL_DATE  LOCAL_HOUR
2015-12-31  0             38
            1             38
            2             38
            3             36
            4             34
2016-01-01  0             38
            1             38
            2             38
            3             36
            4             34
Name: LOCAL_HOUR, dtype: int64

In [16]:
average_wind_speed = df_weather['WIND_SPEED'].mean()
print("Average Wind Speed:", average_wind_speed)

# Maximum Wind Speed
max_wind_speed = df_weather['WIND_SPEED'].max()
print("Maximum Wind Speed:", max_wind_speed)

# Minimum Wind Speed
min_wind_speed = df_weather['WIND_SPEED'].min()
print("Minimum Wind Speed:", min_wind_speed)

Average Wind Speed: 12.492562203081478
Maximum Wind Speed: 114.0
Minimum Wind Speed: 0.0


In [17]:
wind_storm = range(89, 102)

# March 03/08/2017 See Wind Storm
# 04/04/2018
"""

strong_breeze = range(39, 49)
near_gale = range(50, 61)
gale = range(62, 74)
strong_gale = range(75, 88)
strong_breeze_days = df_weather[(df_weather['WIND_SPEED'] >= 39) & (
    df_weather['WIND_SPEED'] <= 49)]
near_gale_days = df_weather[(df_weather['WIND_SPEED'] > 50) & (
    df_weather['WIND_SPEED'] < 61)]
gale_days = df_weather[(df_weather['WIND_SPEED'] > 62) & (
    df_weather['WIND_SPEED'] < 74)]
strong_gale_days = df_weather[(df_weather['WIND_SPEED'] > 75) & (
    df_weather['WIND_SPEED'] < 88)]

strong_breeze_days.head()
near_gale_days.head()
gale_days.head()
strong_gale_days.head()
 """

wind_storm_days_70 = df_weather[(df_weather['WIND_SPEED'] >= 70)]


wind_storm_days_70.head()


id     type  \
24350   6049443.2018.9.21.11  Feature   
24650   6049443.2018.10.3.23  Feature   
24651    6049443.2018.10.4.0  Feature   
116230   6114295.2018.5.4.17  Feature   
126461   613F606.2018.1.2.12  Feature   

                                     coordinates type.1  STATION_PRESSURE  \
24350   [-89.11944444444444, 48.369166666666665]  Point             97.69   
24650   [-89.11944444444444, 48.369166666666665]  Point             97.24   
24651   [-89.11944444444444, 48.369166666666665]  Point             97.60   
116230  [-79.21666666666667, 44.547222222222224]  Point             97.17   
126461               [-79.25, 42.86666666666667]  Point            100.23   

        WINDCHILL_FLAG  UTC_DAY          STATION_NAME  HUMIDEX_FLAG  \
24350              NaN       21  WELCOME ISLAND (AUT)           NaN   
24650              NaN        4  WELCOME ISLAND (AUT)           NaN   
24651              NaN        4  WELCOME ISLAND (AUT)           NaN   
116230             NaN        4           LAGOON CITY           NaN   
126461             NaN        2   PORT COLBORNE (AUT)           NaN   

        WIND_SPEED  VISIBILITY  DEW_POINT_TEMP                  UTC_DATE  \
24350         83.0         NaN             5.5 2018-09-21 16:00:00+00:00   
24650         84.0         NaN             4.4 2018-10-04 04:00:00+00:00   
24651         78.0         NaN             1.5 2018-10-04 05:00:00+00:00   
116230        71.0         NaN             6.9 2018-05-04 22:00:00+00:00   
126461        71.0         NaN           -10.5 2018-01-02 17:00:00+00:00   

        HUMIDEX WIND_SPEED_FLAG  UTC_YEAR  LOCAL_DAY  RELATIVE_HUMIDITY  \
24350       NaN             NaN      2018         21               79.0   
24650       NaN             NaN      2018          3               75.0   
24651       NaN             NaN      2018          4               76.0   
116230      NaN             NaN      2018          4               74.0   
126461      NaN             NaN      2018          2               67.0   

       PROVINCE_CODE  LOCAL_HOUR CLIMATE_IDENTIFIER VISIBILITY_FLAG  TEMP  \
24350             ON          11            6049443             NaN   8.9   
24650             ON          23            6049443             NaN   8.5   
24651             ON           0            6049443             NaN   5.3   
116230            ON          17            6114295             NaN  11.3   
126461            ON          12            613F606             NaN  -5.4   

        UTC_MONTH STATION_PRESSURE_FLAG WIND_DIRECTION_FLAG  \
24350           9                   NaN                 NaN   
24650          10                   NaN                 NaN   
24651          10                   NaN                 NaN   
116230          5                   NaN                 NaN   
126461          1                   NaN                 NaN   

       PRECIP_AMOUNT_FLAG WEATHER_ENG_DESC DEW_POINT_TEMP_FLAG  LOCAL_DATE  \
24350                   M              NaN                 NaN  2018-09-21   
24650                   M              NaN                 NaN  2018-10-04   
24651                   M              NaN                 NaN  2018-10-04   
116230                  M              NaN                 NaN  2018-05-04   
126461                  M              NaN                 NaN  2018-01-02   

        PRECIP_AMOUNT WEATHER_FRE_DESC TEMP_FLAG  LOCAL_YEAR  \
24350             NaN               ND       NaN        2018   
24650             NaN               ND       NaN        2018   
24651             NaN               ND       NaN        2018   
116230            NaN               ND       NaN        2018   
126461            NaN               ND       NaN        2018   

       RELATIVE_HUMIDITY_FLAG                    ID  WINDCHILL  LOCAL_MONTH  \
24350                     NaN  6049443.2018.9.21.11        NaN            9   
24650                     NaN  6049443.2018.10.3.23        NaN           10   
24651                     NaN   6049443.2018.10.

# Found on May 4 2018  >= 80 km/h

- PORT COLBORNE (AUT) - April 4, 2018

    At 14:00, the temperature was 1.3°C, indicative of a chilly day.
    The wind speed reached 81 km/h, suggesting strong wind conditions.

- PA KING CITY - April 16, 2018

    On this day, temperatures were slightly higher but still cool, recorded at 1.7°C at 13:00 and 2.1°C at 14:00.
    The wind speeds were significantly high, especially at 14:00 with a remarkable speed of 114 km/h. This indicates extremely windy conditions, possibly associated with a specific weather event.

In [106]:
wind_storm_days_65 = df_weather[(df_weather['WIND_SPEED'] >= 80 )]
check_may = wind_storm_days_65[(wind_storm_days_65['LOCAL_DATE'] >= '2018-04-01') & (wind_storm_days_65['LOCAL_DATE'] <= '2018-04-31')]
check_may = check_may.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates', 'TEMP'])['WIND_SPEED'].max().reset_index()
check_may

LOCAL_DATE  LOCAL_HOUR         STATION_NAME  \
0  2018-04-04          14  PORT COLBORNE (AUT)   
1  2018-04-16          13         PA KING CITY   
2  2018-04-16          14         PA KING CITY   

                       coordinates  TEMP  WIND_SPEED  
0      [-79.25, 42.86666666666667]   1.3        81.0  
1  [-79.57271944444445, 43.964025]   1.7        91.0  
2  [-79.57271944444445, 43.964025]   2.1       114.0

# Incident September 21, 2018 >= 50 km/h 
 
  > Caribou island (AUT)
    Temperatures ranged from 11.9°C to 14.6°C during the day.
    Wind speeds varied between 72 km/h and 74 km/h, indicating strong winds.

  > WELCOME ISLAND (AUT)

    Recorded a temperature of 8.9°C at 11 AM.
    Experienced a wind speed of 83 km/h, which is notably high.

  > COVE ISLAND (AUT)

    Temperature fluctuated from 14.6°C to 16.3°C.
    Wind speeds were consistent at 73 km/h, suggesting steady strong winds.

  > WESTERN ISLAND (AUT)

    Observed a more substantial temperature variation, from 13.4°C to 19.3°C.
    Wind speeds ranged from 71 km/h to a peak of 93 km/h, indicating very strong winds, especially in the evening.


> HAMILTON A (on September 27, 2018)

    Recorded a temperature of 14.1°C at 9 AM.
    Notably, this station registered an extremely high wind speed of 111 km/h.


In [102]:
data_sep_2018 = df_weather[(df_weather['LOCAL_DATE'] >= '2018-09-21') & (
    df_weather['LOCAL_DATE'] <= '2018-09-30') & (df_weather['WIND_SPEED'] >= 70)]
data_sep_2018 = data_sep_2018.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates', 'TEMP'])[
    'WIND_SPEED'].max().reset_index()

data_sep_2018

LOCAL_DATE  LOCAL_HOUR          STATION_NAME  \
0   2018-09-21          11  CARIBOU ISLAND (AUT)   
1   2018-09-21          11  WELCOME ISLAND (AUT)   
2   2018-09-21          13  CARIBOU ISLAND (AUT)   
3   2018-09-21          14  CARIBOU ISLAND (AUT)   
4   2018-09-21          18     COVE ISLAND (AUT)   
5   2018-09-21          18  WESTERN ISLAND (AUT)   
6   2018-09-21          19     COVE ISLAND (AUT)   
7   2018-09-21          19  WESTERN ISLAND (AUT)   
8   2018-09-21          20  WESTERN ISLAND (AUT)   
9   2018-09-21          21  WESTERN ISLAND (AUT)   
10  2018-09-21          22  WESTERN ISLAND (AUT)   
11  2018-09-27           9            HAMILTON A   

                                 coordinates  TEMP  WIND_SPEED  
0   [-85.83333333333333, 47.333333333333336]  14.6        74.0  
1   [-89.11944444444444, 48.369166666666665]   8.9        83.0  
2   [-85.83333333333333, 47.333333333333336]  11.9        72.0  
3   [-85.83333333333333, 47.333333333333336]  10.6        72.0  
4   [-81.73333333333333, 45.333333333333336]  16.3        73.0  
5    [-80.36666666666666, 45.03333333333333]  19.3        71.0  
6   [-81.73333333333333, 45.333333333333336]  14.6        73.0  
7    [-80.36666666666666, 45.03333333333333]  17.2        87.0  
8    [-80.36666666666666, 45.03333333333333]  14.9        93.0  
9    [-80.36666666666666, 45.03333333333333]  13.8        80.0  
10   [-80.36666666666666, 45.03333333333333]  13.4        73.0  
11             [-79.935, 43.173611111111114]  14.1       111.0

# Windstorm February 2019

# PORT COLBORNE (AUT)

    Temperature ranged from 2.9°C to -1.1°C.
    Wind speeds were extremely high, starting at 90.0 km/h and peaking at 105.0 km/h.
    This station experienced a gradual decrease in temperature over the period, coupled with consistently strong winds.

# LONG POINT (AUT)

    Temperature varied between 1.8°C and -0.8°C.
    Wind speeds were notably high, reaching a maximum of 101.0 km/h.
    The station showed a steady decline in temperature with very strong winds throughout the day.

# WESTERN ISLAND (AUT)

    Temperature decreased from -1.1°C to -5.9°C.
    Wind speeds ranged from 85.0 km/h to 99.0 km/h, remaining high throughout.
    Marked by a steady drop in temperature and persistent high winds, indicating harsh weather conditions.


In [100]:
data_feb_2019 = df_weather[(df_weather['LOCAL_DATE'] >= '2019-02-21') & (
    df_weather['LOCAL_DATE'] <= '2019-02-25') & (df_weather['WIND_SPEED'] >= 85)]
data_feb_2019 = data_feb_2019.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates', 'TEMP'])[
    'WIND_SPEED'].max().reset_index()

data_feb_2019

LOCAL_DATE  LOCAL_HOUR          STATION_NAME  \
0   2019-02-24          13   PORT COLBORNE (AUT)   
1   2019-02-24          14      LONG POINT (AUT)   
2   2019-02-24          14   PORT COLBORNE (AUT)   
3   2019-02-24          15      LONG POINT (AUT)   
4   2019-02-24          15   PORT COLBORNE (AUT)   
5   2019-02-24          16      LONG POINT (AUT)   
6   2019-02-24          16   PORT COLBORNE (AUT)   
7   2019-02-24          17      LONG POINT (AUT)   
8   2019-02-24          17   PORT COLBORNE (AUT)   
9   2019-02-24          18      LONG POINT (AUT)   
10  2019-02-24          18   PORT COLBORNE (AUT)   
11  2019-02-24          19   PORT COLBORNE (AUT)   
12  2019-02-24          20      LONG POINT (AUT)   
13  2019-02-24          20   PORT COLBORNE (AUT)   
14  2019-02-24          21      LONG POINT (AUT)   
15  2019-02-24          21  WESTERN ISLAND (AUT)   
16  2019-02-24          22   PORT COLBORNE (AUT)   
17  2019-02-24          22  WESTERN ISLAND (AUT)   
18  2019-02-24          23  WESTERN ISLAND (AUT)   
19  2019-02-25           0  WESTERN ISLAND (AUT)   
20  2019-02-25           1  WESTERN ISLAND (AUT)   
21  2019-02-25           2  WESTERN ISLAND (AUT)   
22  2019-02-25           3  WESTERN ISLAND (AUT)   
23  2019-02-25           4  WESTERN ISLAND (AUT)   
24  2019-02-25           5  WESTERN ISLAND (AUT)   
25  2019-02-25           6  WESTERN ISLAND (AUT)   
26  2019-02-25           7  WESTERN ISLAND (AUT)   
27  2019-02-25           8  WESTERN ISLAND (AUT)   
28  2019-02-25           9  WESTERN ISLAND (AUT)   

                                coordinates  TEMP  WIND_SPEED  
0               [-79.25, 42.86666666666667]   2.9        90.0  
1               [-80.05, 42.53333333333333]   1.8       101.0  
2               [-79.25, 42.86666666666667]   2.0       105.0  
3               [-80.05, 42.53333333333333]   1.2        98.0  
4               [-79.25, 42.86666666666667]   1.5       101.0  
5               [-80.05, 42.53333333333333]   0.9        99.0  
6               [-79.25, 42.86666666666667]   0.7        97.0  
7               [-80.05, 42.53333333333333]   0.9        91.0  
8               [-79.25, 42.86666666666667]   1.0        91.0  
9               [-80.05, 42.53333333333333]   0.6        91.0  
10              [-79.25, 42.86666666666667]   1.1        96.0  
11              [-79.25, 42.86666666666667]   0.6        88.0  
12              [-80.05, 42.53333333333333]  -0.8        85.0  
13              [-79.25, 42.86666666666667]  -0.1        91.0  
14              [-80.05, 42.53333333333333]  -0.5        86.0  
15  [-80.36666666666666, 45.03333333333333]  -1.1        85.0  
16              [-79.25, 42.86666666666667]  -1.1        86.0  
17  [-80.36666666666666, 45.03333333333333]  -1.6        89.0  
18  [-80.36666666666666, 45.03333333333333]  -2.7        94.0  
19  [-80.36666666666666, 45.03333333333333]  -2.6        94.0  
20  [-80.36666666666666, 45.03333333333333]  -3.0        92.0  
21  [-80.36666666666666, 45.03333333333333]  -2.9        97.0  
22  [-80.36666666666666, 45.03333333333333]  -3.3        85.0  
23  [-80.36666666666666, 45.03333333333333]  -3.4        89.0  
24  [-80.36666666666666, 45.03333333333333]  -4.7        99.0  
25  [-80.36666666666666, 45.03333333333333]  -4.6        87.0  
26  [-80.36666666666666, 45.03333333333333]  -5.1        89.0  
27  [-80.36666666666666, 45.03333333333333]  -5.2        86.0  
28  [-80.36666666666666, 45.03333333333333]  -5.9        86.0

# Dorian September 2019

- the highest only 63 kmh WESTERN ISLAND (AUT) around 5 - 6 am

In [98]:
data_sep_2019 = df_weather[(df_weather['LOCAL_DATE'] >= '2019-09-01') & (
    df_weather['LOCAL_DATE'] <= '2019-09-10') & (df_weather['WIND_SPEED'] >= 60)]
data_sep_2019 = data_sep_2019.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates', 'TEMP'])[
    'WIND_SPEED'].max().reset_index()

data_sep_2019

LOCAL_DATE  LOCAL_HOUR          STATION_NAME  \
0  2019-09-04           5  WESTERN ISLAND (AUT)   
1  2019-09-04           6  WESTERN ISLAND (AUT)   

                               coordinates  TEMP  WIND_SPEED  
0  [-80.36666666666666, 45.03333333333333]  16.2        63.0  
1  [-80.36666666666666, 45.03333333333333]  15.6        61.0

# Ontario Wind Rain 2020

    COVE ISLAND (AUT)
        Temperature ranged from -7.8°C to -8.5°C.
        Wind Speed varied from 65.0 km/h to 68.0 km/h.
        This station experienced consistently cold temperatures with a slight decrease over time, coupled with moderate wind speeds.

    PORT COLBORNE (AUT)
        Temperature ranged from 1.1°C to 4.2°C.
        Wind Speed varied from 71.0 km/h to 89.0 km/h.
        This station had above freezing temperatures with a gradual decrease over time. The wind speeds were high, peaking at 89.0 km/h early on January 12.

In [96]:
data_on_2020 = df_weather[(df_weather['LOCAL_DATE'] >= '2020-01-09') & (
    df_weather['LOCAL_DATE'] <= '2020-01-12') & (df_weather['WIND_SPEED'] >= 65)]
data_on_2020 = data_on_2020.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates', 'TEMP'])[
    'WIND_SPEED'].max().reset_index()

data_on_2020

LOCAL_DATE  LOCAL_HOUR         STATION_NAME  \
0  2020-01-11          23    COVE ISLAND (AUT)   
1  2020-01-12           1    COVE ISLAND (AUT)   
2  2020-01-12           2    COVE ISLAND (AUT)   
3  2020-01-12           5  PORT COLBORNE (AUT)   
4  2020-01-12           6  PORT COLBORNE (AUT)   
5  2020-01-12           7  PORT COLBORNE (AUT)   
6  2020-01-12           8  PORT COLBORNE (AUT)   

                                coordinates  TEMP  WIND_SPEED  
0  [-81.73333333333333, 45.333333333333336]  -7.8        65.0  
1  [-81.73333333333333, 45.333333333333336]  -8.1        68.0  
2  [-81.73333333333333, 45.333333333333336]  -8.5        68.0  
3               [-79.25, 42.86666666666667]   4.2        86.0  
4               [-79.25, 42.86666666666667]   2.5        89.0  
5               [-79.25, 42.86666666666667]   2.1        73.0  
6               [-79.25, 42.86666666666667]   1.1        71.0

# Nov 15 2020
- The highest wind speed was 98 km/h at 21:00 on 15 November at PORT COLOBORNE.

    # PORT COLBORNE (AUT)
- Minimum Wind Speed: 81.0 km/h
- Median Wind Speed: 88.5 km/h
- Maximum Wind Speed: 98.0 km/h


    # WESTERN ISLAND (AUT)
        Minimum Wind Speed: 80.0 km/h
        Median Wind Speed: 84.0 km/h
        Maximum Wind Speed: 87.0 km/h

In [93]:
data_nov_2020 = df_weather[(df_weather['LOCAL_DATE'] >= '2020-11-14') & (
    df_weather['LOCAL_DATE'] <= '2020-11-17') & (df_weather['WIND_SPEED'] >= 80)]
data_nov_2020 = data_nov_2020.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates'])[
    'WIND_SPEED'].max().reset_index()

data_nov_2020

LOCAL_DATE  LOCAL_HOUR          STATION_NAME  \
0   2020-11-15          15   PORT COLBORNE (AUT)   
1   2020-11-15          16   PORT COLBORNE (AUT)   
2   2020-11-15          17   PORT COLBORNE (AUT)   
3   2020-11-15          17  WESTERN ISLAND (AUT)   
4   2020-11-15          18   PORT COLBORNE (AUT)   
5   2020-11-15          18  WESTERN ISLAND (AUT)   
6   2020-11-15          19   PORT COLBORNE (AUT)   
7   2020-11-15          19  WESTERN ISLAND (AUT)   
8   2020-11-15          20   PORT COLBORNE (AUT)   
9   2020-11-15          20  WESTERN ISLAND (AUT)   
10  2020-11-15          21   PORT COLBORNE (AUT)   
11  2020-11-15          22   PORT COLBORNE (AUT)   
12  2020-11-15          22  WESTERN ISLAND (AUT)   
13  2020-11-15          23   PORT COLBORNE (AUT)   
14  2020-11-16           0   PORT COLBORNE (AUT)   
15  2020-11-16           0  WESTERN ISLAND (AUT)   
16  2020-11-16           2  WESTERN ISLAND (AUT)   

                                coordinates  WIND_SPEED  
0               [-79.25, 42.86666666666667]        83.0  
1               [-79.25, 42.86666666666667]        92.0  
2               [-79.25, 42.86666666666667]        95.0  
3   [-80.36666666666666, 45.03333333333333]        82.0  
4               [-79.25, 42.86666666666667]        92.0  
5   [-80.36666666666666, 45.03333333333333]        84.0  
6               [-79.25, 42.86666666666667]        90.0  
7   [-80.36666666666666, 45.03333333333333]        86.0  
8               [-79.25, 42.86666666666667]        82.0  
9   [-80.36666666666666, 45.03333333333333]        84.0  
10              [-79.25, 42.86666666666667]        98.0  
11              [-79.25, 42.86666666666667]        87.0  
12  [-80.36666666666666, 45.03333333333333]        87.0  
13              [-79.25, 42.86666666666667]        81.0  
14              [-79.25, 42.86666666666667]        83.0  
15  [-80.36666666666666, 45.03333333333333]        80.0  
16  [-80.36666666666666, 45.03333333333333]        80.0

# Barrie Tornado
- the wind speed not that high but mostly on the thunderstorms and moderate rain all the area

In [107]:
data_barrie_2021 = df_weather[(df_weather['LOCAL_DATE'] >= '2021-07-14') & (
    df_weather['LOCAL_DATE'] <= '2021-07-17') & (df_weather['WIND_SPEED'] >= 40)]
data_barrie_2021 = data_barrie_2021.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates', 'TEMP'])[
    'WIND_SPEED'].max().reset_index()

data_barrie_2021

LOCAL_DATE  LOCAL_HOUR         STATION_NAME                  coordinates  \
0  2021-07-14           0  PORT COLBORNE (AUT)  [-79.25, 42.86666666666667]   
1  2021-07-14          23  PORT COLBORNE (AUT)  [-79.25, 42.86666666666667]   
2  2021-07-15          21  PORT COLBORNE (AUT)  [-79.25, 42.86666666666667]   
3  2021-07-17           9  PORT COLBORNE (AUT)  [-79.25, 42.86666666666667]   
4  2021-07-17          12    PORT WELLER (AUT)  [-79.21666666666667, 43.25]   
5  2021-07-17          13    PORT WELLER (AUT)  [-79.21666666666667, 43.25]   
6  2021-07-17          14    PORT WELLER (AUT)  [-79.21666666666667, 43.25]   
7  2021-07-17          15  PORT COLBORNE (AUT)  [-79.25, 42.86666666666667]   

   TEMP  WIND_SPEED  
0  21.2        42.0  
1  21.8        46.0  
2  23.4        41.0  
3  17.0        49.0  
4  18.6        42.0  
5  19.0        43.0  
6  19.2        44.0  
7  17.6        42.0

# Ontario Storm 2021 
- the highest wind speed is 65 at PORT COLBN station around 5 - 6 am
- the higer wind speed is 61 km/h found on SARNIA station around 18 - 19 pm
- and all the area is describe as rain and thunderstorms

In [108]:
ontario_storm_2021 = df_weather[(df_weather['LOCAL_DATE'] >= '2021-09-19') & (
    df_weather['LOCAL_DATE'] <= '2021-09-25') & (df_weather['WIND_SPEED'] >= 60)]
ontario_storm_2021 = ontario_storm_2021.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates'])[
    'WIND_SPEED'].max().reset_index()

ontario_storm_2021

LOCAL_DATE  LOCAL_HOUR         STATION_NAME  \
0  2021-09-22          18               SARNIA   
1  2021-09-22          19               SARNIA   
2  2021-09-24           5  PORT COLBORNE (AUT)   
3  2021-09-24           6  PORT COLBORNE (AUT)   
4  2021-09-24           7  PORT COLBORNE (AUT)   

                               coordinates  WIND_SPEED  
0  [-82.30888888888889, 42.99944444444444]        61.0  
1  [-82.30888888888889, 42.99944444444444]        61.0  
2              [-79.25, 42.86666666666667]        60.0  
3              [-79.25, 42.86666666666667]        65.0  
4              [-79.25, 42.86666666666667]        62.0

# Saskatcheman Hall 
- Wind speed showing only 66 km/h at LITTLE FLATLAND ISLAND station
- but the weather is heavy rain with Thunderstorms

In [109]:
ss_storm_2021 = df_weather[(df_weather['LOCAL_DATE'] >= '2021-08-28') & (
    df_weather['LOCAL_DATE'] <= '2021-09-01') & (df_weather['WIND_SPEED'] >= 50)]
ss_storm_2021 = ss_storm_2021.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates'])[
    'WIND_SPEED'].max().reset_index()

ss_storm_2021

LOCAL_DATE  LOCAL_HOUR                  STATION_NAME  \
0  2021-08-29          14          WELCOME ISLAND (AUT)   
1  2021-08-29          16  LITTLE FLATLAND ISLAND (AUT)   
2  2021-08-29          17  LITTLE FLATLAND ISLAND (AUT)   

                                coordinates  WIND_SPEED  
0  [-89.11944444444444, 48.369166666666665]        50.0  
1               [-88.30583333333334, 49.69]        66.0  
2               [-88.30583333333334, 49.69]        61.0

# Ontario Storm on December 2021 
- The highest wind speed is 106 km/h at PORT COLBORNE aorund 16 - 18 
- Heavy Storm


In [81]:
dec_on_storm_2021 = df_weather[(df_weather['LOCAL_DATE'] >= '2021-12-11') & (
    df_weather['LOCAL_DATE'] <= '2021-12-15') & (df_weather['WIND_SPEED'] >= 80)]
dec_on_storm_2021 = dec_on_storm_2021.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates'])[
    'WIND_SPEED'].max().reset_index()

dec_on_storm_2021

LOCAL_DATE  LOCAL_HOUR          STATION_NAME  \
0   2021-12-11          15   PORT COLBORNE (AUT)   
1   2021-12-11          16   PORT COLBORNE (AUT)   
2   2021-12-11          17     POINT PETRE (AUT)   
3   2021-12-11          17   PORT COLBORNE (AUT)   
4   2021-12-11          18     POINT PETRE (AUT)   
5   2021-12-11          18   PORT COLBORNE (AUT)   
6   2021-12-11          18  WESTERN ISLAND (AUT)   
7   2021-12-11          19     POINT PETRE (AUT)   
8   2021-12-11          19   PORT COLBORNE (AUT)   
9   2021-12-11          19  WESTERN ISLAND (AUT)   
10  2021-12-11          20  WESTERN ISLAND (AUT)   

                                coordinates  WIND_SPEED  
0               [-79.25, 42.86666666666667]        93.0  
1               [-79.25, 42.86666666666667]       106.0  
2              [-77.15, 43.833333333333336]        91.0  
3               [-79.25, 42.86666666666667]       105.0  
4              [-77.15, 43.833333333333336]        99.0  
5               [-79.25, 42.86666666666667]        84.0  
6   [-80.36666666666666, 45.03333333333333]        88.0  
7              [-77.15, 43.833333333333336]        86.0  
8               [-79.25, 42.86666666666667]        88.0  
9   [-80.36666666666666, 45.03333333333333]        84.0  
10  [-80.36666666666666, 45.03333333333333]        85.0

# Ontario Flooding

- PORT COLBORNE (AUT)

    During the early hours (02:00 to 06:00), the temperatures hovered around -0.7°C to -1.8°C, indicating a cold environment.
    Wind speeds were consistently high, ranging from 75 km/h to 78 km/h, suggesting strong and possibly gusty conditions.
    By 13:00, the temperature dropped significantly to -7.2°C, with wind speeds remaining strong at 70 km/h.

- WESTERN ISLAND (AUT) - 05:00

    The temperature was recorded at -4.1°C, colder compared to PORT COLBORNE.
    Wind speed was 73 km/h, indicating similarly strong winds.
    POINT PETRE (AUT) - 06:00
    The temperature was -1.1°C, aligning with the cold conditions observed in other locations.
    Wind speed was slightly lower at 70 km/h but still indicative of a breezy environment.

In [111]:
on_flooding = df_weather[(df_weather['LOCAL_DATE'] >= '2022-02-13') & (
    df_weather['LOCAL_DATE'] <= '2022-02-22') & (df_weather['WIND_SPEED'] >= 70)]
on_flooding = on_flooding.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates', 'TEMP'])[
    'WIND_SPEED'].max().reset_index()
on_flooding

LOCAL_DATE  LOCAL_HOUR          STATION_NAME  \
0  2022-02-19           2   PORT COLBORNE (AUT)   
1  2022-02-19           3   PORT COLBORNE (AUT)   
2  2022-02-19           4   PORT COLBORNE (AUT)   
3  2022-02-19           5   PORT COLBORNE (AUT)   
4  2022-02-19           5  WESTERN ISLAND (AUT)   
5  2022-02-19           6     POINT PETRE (AUT)   
6  2022-02-19           6   PORT COLBORNE (AUT)   
7  2022-02-19          13   PORT COLBORNE (AUT)   

                               coordinates  TEMP  WIND_SPEED  
0              [-79.25, 42.86666666666667]  -0.7        75.0  
1              [-79.25, 42.86666666666667]  -0.9        77.0  
2              [-79.25, 42.86666666666667]  -0.8        78.0  
3              [-79.25, 42.86666666666667]  -1.8        76.0  
4  [-80.36666666666666, 45.03333333333333]  -4.1        73.0  
5             [-77.15, 43.833333333333336]  -1.1        70.0  
6              [-79.25, 42.86666666666667]  -0.9        78.0  
7              [-79.25, 42.86666666666667]  -7.2        70.0

# MB / NW ont flooding

- all the area wind speed around 40 but one indicate as 50 km/h is Toronto Downtown
- all the area was snowing and some is heavy snow
- one area is more than 69 is WESTERN ISLAND (AUT)	

In [113]:
mb_flooding = df_weather[(df_weather['LOCAL_DATE'] >= '2022-04-20') & (
    df_weather['LOCAL_DATE'] <= '2022-04-30') & (df_weather['WIND_SPEED'] >= 50)]
mb_flooding = mb_flooding.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'coordinates', 'STATION_NAME'])[
    'WIND_SPEED'].max().reset_index()
mb_flooding

LOCAL_DATE  LOCAL_HOUR                               coordinates  \
0  2022-04-20          19  [-85.83333333333333, 47.333333333333336]   
1  2022-04-20          22  [-85.83333333333333, 47.333333333333336]   
2  2022-04-21          14               [-79.25, 42.86666666666667]   
3  2022-04-23           6  [-89.11944444444444, 48.369166666666665]   
4  2022-04-23          11             [-79.39611111111111, 43.6275]   
5  2022-04-23          12   [-80.36666666666666, 45.03333333333333]   
6  2022-04-23          20             [-79.39611111111111, 43.6275]   
7  2022-04-24           3  [-85.83333333333333, 47.333333333333336]   
8  2022-04-24           4  [-85.83333333333333, 47.333333333333336]   
9  2022-04-24           6  [-85.83333333333333, 47.333333333333336]   

           STATION_NAME  WIND_SPEED  
0  CARIBOU ISLAND (AUT)        51.0  
1  CARIBOU ISLAND (AUT)        50.0  
2   PORT COLBORNE (AUT)        51.0  
3  WELCOME ISLAND (AUT)        54.0  
4   TORONTO CITY CENTRE        50.0  
5  WESTERN ISLAND (AUT)        70.0  
6   TORONTO CITY CENTRE        54.0  
7  CARIBOU ISLAND (AUT)        50.0  
8  CARIBOU ISLAND (AUT)        51.0  
9  CARIBOU ISLAND (AUT)        50.0

# Ontario Storm on May2022 21/22

- the highest wind speed is 60 km/h
- Heavy Rain and Thunderstorms

In [116]:
may_2022_storm = df_weather[(df_weather['LOCAL_DATE'] >= '2022-05-19') & (
    df_weather['LOCAL_DATE'] <= '2022-05-25') & (df_weather['WIND_SPEED'] >= 50)]
may_2022_storm = may_2022_storm.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates', 'TEMP'])[
    'WIND_SPEED'].max().reset_index()
may_2022_storm

LOCAL_DATE  LOCAL_HOUR    STATION_NAME  \
0  2022-05-21          10    CHATHAM KENT   
1  2022-05-21          12  TORONTO INTL A   

                               coordinates  TEMP  WIND_SPEED  
0            [-82.0825, 42.30583333333333]  22.3        55.0  
1  [-79.63055555555556, 43.67666666666667]  17.0        60.0

# Hurrican Fiona

- Rain some area and heavy rain
- but on Toronto is Cloudy but quite low temperature
- wind speed around 64 km/h

In [119]:
fiona = df_weather[(df_weather['LOCAL_DATE'] >= '2022-09-20') & (
    df_weather['LOCAL_DATE'] <= '2022-09-26') & (df_weather['WIND_SPEED'] >= 60)]
fiona = fiona.groupby(['LOCAL_DATE', 'LOCAL_HOUR', 'STATION_NAME', 'coordinates'])[
    'WIND_SPEED'].max().reset_index()
fiona

LOCAL_DATE  LOCAL_HOUR          STATION_NAME  \
0  2022-09-21          15  CARIBOU ISLAND (AUT)   
1  2022-09-21          21  WESTERN ISLAND (AUT)   
2  2022-09-26          12   PORT COLBORNE (AUT)   
3  2022-09-26          22     POINT PETRE (AUT)   

                                coordinates  WIND_SPEED  
0  [-85.83333333333333, 47.333333333333336]        61.0  
1   [-80.36666666666666, 45.03333333333333]        64.0  
2               [-79.25, 42.86666666666667]        62.0  
3              [-77.15, 43.833333333333336]        64.0

In [75]:

df_weather['WIND_SPEED'] = pd.to_numeric(df_weather['WIND_SPEED'], errors='coerce')
grouped_max_wind_speed = df_weather.groupby(['STATION_NAME', 'coordinates', 'LOCAL_DATE', 'LOCAL_HOUR'])['WIND_SPEED'].max()

top_5 = grouped_max_wind_speed.nlargest(5)
top_5


STATION_NAME         coordinates                      LOCAL_DATE  LOCAL_HOUR
PA KING CITY         [-79.57271944444445, 43.964025]  2018-04-16  14            114.0
HAMILTON A           [-79.935, 43.173611111111114]    2018-09-27  9             111.0
PORT COLBORNE (AUT)  [-79.25, 42.86666666666667]      2021-12-11  16            106.0
                                                      2019-02-24  14            105.0
                                                      2021-12-11  17            105.0
Name: WIND_SPEED, dtype: float64